rviz_marker_tools/src/marker_creation.cpp

```diff
@@ -276,6 +276,22 @@ vm::Marker& makeMesh(vm::Marker& m, const std::string& filename, double sx, doub
 	return m;
 }
 
+vm::Marker& makeArrow(vm::Marker& m, const Eigen::Vector3d& start_point, const Eigen::Vector3d& end_point, double diameter, double head_length) {
+	// scale.y is set according to default proportions in rviz/default_plugin/markers/arrow_marker.cpp#L61
+	// for the default head_length=0, the head length will keep the default proportion defined in arrow_marker.cpp#L106
+	m.scale.x = diameter;
+	m.scale.y = 2 * diameter;
+	m.scale.z = head_length;
+	prepareMarker(m, vm::Marker::ARROW);
```

Comment:
```suggestion
	prepareMarker(m, vm::Marker::ARROW);
	// scale.y is set according to default proportions in rviz/default_plugin/markers/arrow_marker.cpp#L61
	// for the default head_length=0, the head length will keep the default proportion defined in arrow_marker.cpp#L106
	m.scale.x = diameter;
	m.scale.y = 2 * diameter;
	m.scale.z = head_length;

Reply:
I put `prepareMarker` after the `scale` assignments, because it ensures a valid scale. Currently, this is not necessary, as `diameter` and `head_length` are derived from `linear_norm`, which is verified to be non-zero for the visualization.

However, if this should ever change in the future, I thought it's better to have that scale check, especially since all other methods call `prepareMarker` after the `scale` assignments to ensure a valid scale.